data-prepper-plugins/kinesis-source/src/main/java/org/opensearch/dataprepper/plugins/kinesis/source/KinesisClientApiHandler.java

```diff
@@ -77,13 +120,18 @@ private void applyBackoff() {
         try {
             Thread.sleep(delayMillis);
         } catch (final InterruptedException e){
+            Thread.currentThread().interrupt();
```

Comment:
You reset the interrupt here. But, does anything actually monitor the interrupt status? Maybe you should end the backoff/retry loop.